remoting/getty/getty_client.go

```diff
@@ -204,6 +204,14 @@ func (c *Client) Request(request *remoting.Request, timeout time.Duration, respo
 	return perrors.WithStack(err)
 }
 
+func (c *Client) IsAvailable() bool {
+	client, err := c.pool.get()
+	if client == nil || !client.isAvailable() || err != nil {
+		return false
+	}
+	return true
```

Comment:
```suggestion
	return  client != nil && client.isAvailable() && err == nil 
```

Reply:
I have changed the implementation to 

```go
// isAvailable returns true if the connection is available, or it can be re-established.
func (c *Client) IsAvailable() bool {
	_, _, err := c.selectSession(c.addr)
	return err == nil
}
```

since we need to re-establish the connection if possible.